### PR TITLE
Drop GA-ed or deprecated feature gates

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area open-source
/kind cleanup

**What this PR does / why we need it**:
- Drop GA-ed `ShootMaxTokenExpiration{Overwrite,Validation}` feature gates
- Drop deprecated `RotateSSHKeypairOnMaintenance` feature gate
- Drop description for no longer existing feature gates

**Release note**:

```breaking operator
The GA-ed or deprecated `ShootMaxTokenExpiration{Overwrite,Validation}` and `RotateSSHKeypairOnMaintenance` feature gates have been removed.
```

### CODE REVIEW

File: pkg/registry/core/shoot/strategy.go [hunk 1]
```diff
@@ -80,28 +75,6 @@ func (shootStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Obje
...
 	}
-
-	if utilfeature.DefaultFeatureGate.Enabled(features.ShootMaxTokenExpirationOverwrite) {
-		// Note that this must be executed after `mustIncreaseGeneration` is called, otherwise we might trigger a
-		// reconciliation outside of the maintenance time window because the spec might change in below call.
-		overwriteMaxTokenExpiration(newShoot)
-	}
-}
-
-func overwriteMaxTokenExpiration(shoot *core.Shoot) {
```
Comment (ary1992) [inline pkg/registry/core/shoot/strategy.go:91]: Why this function is removed? Is the `overwriteMaxTokenExpiration ()` function no longer required ?

File: docs/deployment/feature_gates.md [hunk 2]
```diff
@@ -133,21 +133,14 @@ A *General Availability* (GA) feature is also referred to as a *stable* feature.
...
 | ForceRestore                               | `gardenlet`                                                      | Enables forcing the shoot's restoration to the destination seed during control plane migration if the preparation for migration in the source seed is not finished after a certain grace period and is considered unlikely to succeed (falling back to the [control plane migration "bad case" scenario](../proposals/17-shoot-control-plane-migration-bad-case.md)). If you enable this feature gate, make sure to also enable `UseDNSRecords` and `CopyEtcdBackupsDuringControlPlaneMigration`. |
 | DisableDNSProviderManagement               | `gardenlet`                                                      | Disables management of `dns.gardener.cloud/v1alpha1.DNSProvider` resources. In this case, the `shoot-dns-service` extension will take this over if it is installed. This feature is only effective if the feature `UseDNSRecords` is `true`. |
```
Comment (ary1992) [inline docs/deployment/feature_gates.md]: As the description of `UseDNSRecords` is removed , we can adapt it here as well. WDYT ? 
```suggestion
| DisableDNSProviderManagement               | `gardenlet`                                                      | Disables management of `dns.gardener.cloud/v1alpha1.DNSProvider` resources. In this case, the `shoot-dns-service` extension will take this over if it is installed. |
```

File: docs/deployment/feature_gates.md [hunk 3]
```diff
@@ -133,21 +133,14 @@ A *General Availability* (GA) feature is also referred to as a *stable* feature.
...
 | SecretBindingProviderValidation            | `gardener-apiserver`                                             | Enables validations on Gardener API server that:<br>- requires the provider type of a SecretBinding to be set (on SecretBinding creation)<br>- requires the SecretBinding provider type to match the Shoot provider type (on Shoot creation)<br>- enforces immutability on the provider type of a SecretBinding |
 | ForceRestore                               | `gardenlet`                                                      | Enables forcing the shoot's restoration to the destination seed during control plane migration if the preparation for migration in the source seed is not finished after a certain grace period and is considered unlikely to succeed (falling back to the [control plane migration "bad case" scenario](../proposals/17-shoot-control-plane-migration-bad-case.md)). If you enable this feature gate, make sure to also enable `UseDNSRecords` and `CopyEtcdBackupsDuringControlPlaneMigration`. |
```
Comment (ary1992) [inline docs/deployment/feature_gates.md]:
```suggestion
| ForceRestore                               | `gardenlet`                                                      | Enables forcing the shoot's restoration to the destination seed during control plane migration if the preparation for migration in the source seed is not finished after a certain grace period and is considered unlikely to succeed (falling back to the [control plane migration "bad case" scenario](../proposals/17-shoot-control-plane-migration-bad-case.md)). If you enable this feature gate, make sure to also enable `CopyEtcdBackupsDuringControlPlaneMigration`. |
```

**Review by ary1992 — COMMENTED** [review]
Thank you for the PR.

**Comment by shafeeqes** [comment]
Please adapt https://github.com/gardener/gardener/blob/23267e48edf86a497b2e196f45bae7f40f993d7e/docs/proposals/17-shoot-control-plane-migration-bad-case.md#L13
this line as well.

File: pkg/registry/core/shoot/strategy.go (hunk 1 above)
Comment (rfranzke) [reply pkg/registry/core/shoot/strategy.go:91]: Yes, it was only required temporarily for overwriting this field for existing shoot clusters.

**Pull request merged**